.travis/download_and_symlink_z3.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+set -ev
+
+mkdir -p ${DOWNLOADS_DIR}
```

Comment:
Add

```bash
: ${DOWNLOADS_DIR?"DOWNLOADS_DIR must be set}
: ${Z3?"Z3 must be set to???"}
```

Again it would be better to not depend on TRAVIS_BUILD_DIR here.